LanguageFeatures/Extension-types/exhaustiveness_enum_A01_t01.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with enum as a matched type are
+/// always exhaustive
+///
+/// @description Check that it is no compile-time error if a matched type of a
+/// switch expression is an extension type with an enum as a representation type
+/// and the set of cases is exhaustive
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+enum E<T> {
+  a<int>(),
+  b<String>(),
+  c<double>(),
+}
+
+extension type ET1<T>(E<T> _) {}
+extension type ET2<T>(E<T> _) implements E<T> {}
+
+String testStatement1(ET1<num> e) {
+  switch (e) {
+    case E.a:
+    case E.c:
+      return "ok";
```

Comment:
This seems to work, but it would be somewhat esoteric (and this has nothing to do with extension types, the situation exists after erasure).

The reason why this is a very special exception is that we (presumably) rely on the fact that this is a switch over enumerated values. This means that we can iterate over all the elements, and it is _guaranteed_ that we won't encounter any other objects than the ones that are declared in the `enum` declaration itself, and they have a statically known run-time type.

Lots and lots of stuff to know!

OK, I was worried that this would be on the very edge of the capabilities of the exhaustiveness analysis, but it seems to work fine.

In any case, we should probably also have a simpler test where the enumerated type isn't generic, such that it is easy to see that we have an extension type on top of a very straightforward setup (which will be a test that is only about being an extension type, so it's very safe to assume that it will test just that).